plugins/georeplication/init.go

```diff
@@ -0,0 +1,46 @@
+package georeplication
+
+import (
+	"github.com/gluster/glusterd2/servers/rest/route"
+	"github.com/gluster/glusterd2/transaction"
+	"github.com/prashanthpai/sunrpc"
+)
+
+// Plugin is a structure which implements GlusterdPlugin interface
+type Plugin struct {
+}
+
+// Name returns name of plugin
+func (p *Plugin) Name() string {
+	return "georeplication"
+}
+
+// SunRPCProgram returns sunrpc program to register with Glusterd
+func (p *Plugin) SunRPCProgram() sunrpc.Program {
+	return nil
+}
+
+// RestRoutes returns list of REST API routes to register with Glusterd
+func (p *Plugin) RestRoutes() route.Routes {
+	return route.Routes{
+		route.Route{
+			Name:        "GeoreplicationCreate",
+			Method:      "PUT",
```

Comment:
Personally, I really prefer `PUT` over `POST` for resource creation and `POST` for resource modification.

But it's been `POST` for all requests in REST API so far. We should open up this for discussion. We will need the APIs to be uniform i.e creation of volumes, snapshot, georep to be all `PUT` or all `POST`.

Reply:
Also, `mastervolid` and `slaveid` can also be part of request body. Any reason for it being part of URL ?

Reply:
> Also, mastervolid and slaveid can also be part of request body. Any reason for it being part of URL ?

Since it is unique combination for a georep session and required always. It also makes more sense for other commands start/stop/status/config (Ex: POST /v1/geo-replication/:masterid/:slaveid/start)

Reply:
I will change this to POST